build.sh

```diff
@@ -1,3052 +1,129 @@
-    build-arm-benchmark:
-    build-config-progs:
-    build-osx:
-    build-ios:
-    build-ios-no-bitcode:
-    build-watchos:
-    build-tvos:
-    build-android:
-    build-cocoa:
-    build-dotnet-cocoa:
-    build-osx-framework:
-    build-node:
-    build-node-package:
-    test:
-    test-debug:
-    check:
-    check-debug:
-    memcheck:
-    memcheck-debug:
-    check-testcase:
-    check-testcase-debug:
-    memcheck-testcase:
-    memcheck-testcase-debug:
-    asan:
-    asan-debug:
-    build-test-ios-app:                 build an iOS app for testing core on device
-    test-ios-app:                       execute the core tests on device
-    leak-test-ios-app:                  execute the core tests on device, monitor for leaks
-    gdb:
-    gdb-debug:
-    gdb-testcase:
-    gdb-testcase-debug:
-    performance:
-    benchmark:
-    benchmark-*:
-    lcov:
-    gcovr:
-    show-install:
-    release-notes-prerelease:
-    release-notes-postrelease:
-    get-version:
-    set-version:
-    copy-tools:
-    install:
-    install-prod:
-    install-devel:
-    uninstall:
-    uninstall-prod:
-    uninstall-devel:
-    test-installed:
-    wipe-installed:
-    src-dist:
-    bin-dist:
-    dist-config:
-    dist-clean:
-    dist-build:
-    dist-build-iphone:
-    dist-test:
-    dist-test-debug:
-    dist-install:
-    dist-uninstall:
-    dist-test-installed:
-    dist-status:
-    dist-pull:
-    dist-checkout:
-    dist-copy:
-    jenkins-pull-request:               Run by Jenkins for each pull request whenever it changes
-    jenkins-pipeline-unit-tests:        Run by Jenkins as part of the core pipeline whenever master changes
-    jenkins-pipeline-coverage:          Run by Jenkins as part of the core pipeline whenever master changes
-    jenkins-pipeline-address-sanitizer: Run by Jenkins as part of the core pipeline whenever master changes
-    jenkins-valgrind:
-EOF
-}
-
-map_ext_name_to_dir()
-{
-    local ext_name
-    ext_name="$1"
-    case $ext_name in
-        *) echo "realm_$ext_name";;
+#!/bin/bash
+
+#Set Script Name variable
+SCRIPT=$(basename "${BASH_SOURCE[0]}")
+DIR=$(dirname "${BASH_SOURCE[0]}")
+
+# Set fonts
+NORM="\e[0m"
+BOLD="\e[1m"
+RED="${BOLD}\e[31m"
+
+# Number of cores
+CORES=$(getconf _NPROCESSORS_ONLN)
+
+function usage {
+    echo "${BOLD}Usage:${NORM} ${SCRIPT} -t <build_type> -o <target_os> -v <version> [-a <android_abi>]"
+    echo ""
+    echo "${BOLD}Arguments:${NORM}"
+    echo "   build_type=<Release|Debug|MinSizeDebug>"
+    echo "   target_os=<android|ios|watchos|tvos>"
+    echo "   android_abi=<armeabi|armeabi-v7a|x86|mips|x86_64|arm64-v8a>"
+    exit 1;
+}
+
+# Parse the options
+while getopts ":o:a:t:v:" opt; do
+    case "${opt}" in
+        o)
+            OS=${OPTARG}
+            [ "${OS}" == "android" ] ||
+            [ "${OS}" == "ios" ] ||
+            [ "${OS}" == "watchos" ] ||
+            [ "${OS}" == "tvos" ] || usage
```

Comment:
Better use of `grep` revealed that the Jenkinsfile invokes `xcodebuild` directly multiple times to produce a `.tar.gz`. That seems difficult to use for local development.

Why does the Mac build produce a `.gz` file rather than a `.xz` like we do for other Apple platforms?